plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -420,14 +425,51 @@ private class DokkaDescriptorVisitor(
         }
     }
 
-    private suspend fun visitPropertyDescriptor(originalDescriptor: PropertyDescriptor): DProperty {
-        val (dri, inheritedFrom) = originalDescriptor.createDRI()
+    /**
+     * @param implicitAccessors getters/setters that are not part of the property descriptor, for instance
+     *                          average methods inherited from java sources
+     */
+    private suspend fun visitPropertyDescriptor(
+        originalDescriptor: PropertyDescriptor,
+        implicitAccessors: List<FunctionDescriptor>,
+        parent: DRIWithPlatformInfo
+    ): DProperty {
+        val (dri, _) = originalDescriptor.createDRI()
+        val inheritedFrom = dri.getInheritedFromDRI(parent)
         val descriptor = originalDescriptor.getConcreteDescriptor()
         val isExpect = descriptor.isExpect
         val isActual = descriptor.isActual
 
         val actual = originalDescriptor.createSources()
 
+        // example - generated getter that comes with data classes
+        suspend fun getDescriptorGetter() =
+            descriptor.accessors
+                .filterIsInstance<PropertyGetterDescriptor>()
+                .singleOrNull()
+                ?.let {
+                    visitPropertyAccessorDescriptor(it, descriptor, dri, inheritedFrom)
+                }
+
+        suspend fun getImplicitAccessorGetter() =
+            implicitAccessors
+                .firstOrNull { it.isGetterFor(originalDescriptor) }
```

Comment:
Not sure what you mean unfortunately, but I've had a look at the functions it and changed `singleOrNull` to `firstIsInstanceOrNull` to avoid an intermediate operation, maybe that answers the question. If not, can you please demonstrate?

Reply:
On line 449 you use `singleOrNull`, but `singleOrNull` is used here. What function should be applied?